### PR TITLE
Add org_id column populator job

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-2526](https://issues.redhat.com/browse/ESSNTL-2526).
It adds org-id-populator Job and CJI definitions to our deployment. Documentation for this job can be found [here](https://github.com/RedHatInsights/tenant-utils/blob/main/cmd/org-id-column-populator/README.md).

This should not be merged until:

- #1127 is merged and working
- the appropriate permission configs are added in app-interface ([MR here](https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/40315))

After the job runs, we need to update `rebuild_events_topic.py` to send Update events for all of these hosts and run it.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
ff269aed-66c1-404d-b6b3-6b3334265cfb

File: deployment.yaml [hunk 1]
```diff
@@ -558,6 +558,34 @@ objects:
           requests:
             cpu: ${CPU_REQUEST}
             memory: ${MEMORY_REQUEST}
+    - name: org-id-populator
```
Comment (thearifismail) [inline deployment.yaml:561]: Needs a comment regarding the source of of `org-id-column-populator`; once this PR is merged, readers may wonder about this script provider.

File: deployment.yaml [hunk 2]
```diff
@@ -784,6 +820,14 @@ parameters:
   value: 'apicast.3scale-dev.svc.cluster.local'
 - name: TENANT_TRANSLATOR_PORT
   value: '8892'
+- name: POPULATOR_LOG_FORMAT
+  value: cloudwatch
+- name: POPULATOR_IMAGE
+  value: quay.io/cloudservices/tenant-utils
+- name: POPULATOR_IMAGE_TAG
+  value: latest
```
Comment (thearifismail) [inline deployment.yaml:828]: should us a specific image tag.

File: deployment.yaml (hunk 1 above)
Comment (kruai) [reply deployment.yaml:567]: Done!

File: deployment.yaml (hunk 2 above)
Comment (thearifismail) [reply deployment.yaml:834]: I guess it is an issue against the provider then, but should not block host-inventory

**Pull request merged**